modules/fedimint-mint/src/lib.rs

```diff
@@ -179,9 +179,12 @@ impl ModuleGen for MintGen {
     fn trusted_dealer_gen(
         &self,
         peers: &[PeerId],
+        module_id: ModuleInstanceId,
         params: &ConfigGenParams,
     ) -> BTreeMap<PeerId, ServerModuleConfig> {
-        let params = params.get::<MintGenParams>().expect("Invalid mint params");
+        let params = params
+            .get::<MintGenParams>(&module_id)
```

Comment:
The module should get its own params, and not the whole map of params and it's id to look it up.

Reply:
So pass in the JSON value instead?  This is how we were handling it in distributed_gen.